src/config/wmodules-aws.c

```diff
@@ -418,17 +430,26 @@ int wm_aws_read(const OS_XML *xml, xml_node **nodes, wmodule *module)
                             os_strdup(children[j]->content, cur_service->iam_role_duration);
                         }
                 } else if (!strcmp(children[j]->element, XML_ONLY_LOGS_AFTER)) {
-                    if (strlen(children[j]->content) != 0) {
+                    if (strlen(children[j]->content) == 0) {
+                            mwarn("Empty content for tag '%s' at module '%s'.", XML_ONLY_LOGS_AFTER, WM_AWS_CONTEXT.name);
+                    }
+                    else if (strlen(children[j]->content) != 0) {
                         free(cur_service->only_logs_after);
                         os_strdup(children[j]->content, cur_service->only_logs_after);
                     }
                 } else if (!strcmp(children[j]->element, XML_REGION)) {
-                    if (strlen(children[j]->content) != 0) {
+                    if (strlen(children[j]->content) == 0) {
+                            mwarn("Empty content for tag '%s' at module '%s'.", XML_REGION, WM_AWS_CONTEXT.name);
+                    }
+                    else if (strlen(children[j]->content) != 0) {
                         free(cur_service->regions);
                         os_strdup(children[j]->content, cur_service->regions);
                     }
                 } else if (!strcmp(children[j]->element, XML_LOG_GROUP)) {
-                    if (strlen(children[j]->content) != 0) {
+                    if (strlen(children[j]->content) == 0) {
+                            mwarn("Empty content for tag '%s' at module '%s'.", XML_LOG_GROUP, WM_AWS_CONTEXT.name);
```

Comment:
As stated in the [documentation](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/wodle-s3.html#services), the `aws_log_groups` is a mandatory field for CloudWatch Logs service, therefore an error and exit could be best in this case instead of a warning

Reply:
Good catch, fixed in 175b9a9